### ISSUE TITLE
[2021 Theme Proposal Summary] Improved developer experience

### ISSUE THREAD

**Issue by JonasKruckenberg** [ipfs/roadmap#88]
_Note, this is part of the [2021 IPFS project planning process ](https://github.com/ipfs/roadmap/blob/master/2021-IPFS-Project-Planning.md) - feel free to add other potential 2021 themes for the IPFS project by opening a [new issue](https://github.com/ipfs/roadmap/issues/new/choose) or discuss this proposed theme in the comments, especially other example workstreams that could fit under this theme for 2021. Please also review others’ proposed themes and leave feedback [here](https://github.com/ipfs/roadmap/labels/2021%20Theme%20Proposal)!_
## Theme description
A lot of current proposals ( while very valid in their own right ) focus on improving the experience of IPFS developers. 
This proposal is intended to summarize their commonalities and provide actionable ideas.

## Hypothesis
*Improving the overall IPFS developer experience will provide more and better results faster.*

## Vision statement
A bigger interest in IPFS and it's core ideas will bring new developers to the community that build apps on IPFS, help with software in the stack and more financial investment as well.
Features can be implemented faster and a diverse set of developers represented in the community will lead to mature features.

## Why focus this year
This has been a focus last year and it should stay a focus in 2021.

## Example workstreams
DX proposals and issues over the last year boil down to four main *areas of improvement*:

1.  **APIs**
#62,#61 capture this idea pretty well, improving the APIs and architecture of IPFS implementations helps to onboard new developers and increase adoption.

2. **Tooling**
#87 fits into this, as well as #77 and especially #63.
Refer to each proposal for specifics, but in general this is awesome! Mature tooling would further adoption across the industry since no one likes working with arcane tech for a big project right?

3. **Specifications**
This is another big one, we should work out specifications across the stack. This would make it easier for people to develop compliant implementations and also get the IPFS community aligned.
This is especially pressing since we don't want to end up with a developer aristocracy where the stack has become so complicated and arcane that only a select few can work on the project.
(Looking at you graphsync spec :new_moon_with_face:)

4. **Talk about IPFS!**
Talk to your coworkers, friends and family about this! IPFS is awesome, let them know!
Talking about IPFS also includes a bigger media presence( blog articles etc. ),
as well as conventions ( at some point :| ) and encouraging people to become IPFS developer advocates in their communities.
#86 fits in here quite nicely.
**in short just keep the ecosystem WG around**

## Other content
Refer to all proposals mentioned above for their specific ideas
